File: exercises/tests/necklace_test.js
import test from "./test.js";
import necklace from "../src/necklace.js";

test({
  description: "necklace baseline",
  actual: necklace({
    original: "nicole",
    possibleVariation: "lenico",
  }),
  expected: true,
});

test({
  description: "necklace basic no",
  actual: necklace({
    original: "nicole",
    possibleVariation: "coneli",
  }),
  expected: false,
});

test({
  description: "necklace long",
  actual: necklace({
    original: "aabaaaaabaab",
    possibleVariation: "aabaabaabaaa",
  }),
  expected: true,
});

test({
  description: "necklace single character",
  actual: necklace({
    original: "a",
    possibleVariation: "a",
  }),
  expected: true,
});

test({
  description: "necklace missing",
  actual: necklace({
    original: "a",
    possibleVariation: "",
  }),
  expected: false,
});

test({
  description: "necklace incomparable",
  actual: necklace({
    original: "abc",
    possibleVariation: "abcd",
  }),
  expected: false,
});

test({
  description: "necklace empty",
  actual: necklace({
    original: "",
    possibleVariation: "",
  }),
  expected: true,
});

File: rust/exercises/tests/necklace_test.rs
#[test]
fn same() {
    assert_eq!(exercises::necklace::is_same("nicole", "icolen"), true);
}

#[test]
fn not_same() {
    assert_eq!(exercises::necklace::is_same("nicole", "coneli"), false);
}

#[test]
fn same_long() {
    assert_eq!(
        exercises::necklace::is_same("aabaaaaabaab", "aabaabaabaaa"),
        true
    );
}

#[test]
fn not_same_long() {
    assert_eq!(
        exercises::necklace::is_same("baabaaaaabaab", "aabaabaabaaa"),
        false
    );
}

#[test]
fn single() {
    assert_eq!(exercises::necklace::is_same("x", "x"), true);
}

#[test]
fn mismatch() {
    assert_eq!(exercises::necklace::is_same("x", "xx"), false);
}

#[test]
fn mismatch_empty() {
    assert_eq!(exercises::necklace::is_same("x", ""), false);
}

#[test]
fn empty() {
    assert_eq!(exercises::necklace::is_same("", ""), true);
}
